### PR TITLE
Project submission

### PR DESCRIPTION
**Pull request by Mahamadou-Nouridine**
In this Pr, I brought changes according to the following requirements:

#### Design
- You should follow these [design guidelines](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), including:
  - Colors.
  - Typography: font face, size and weight.
  - Layout: composition and space between elements.

> NOTE: In these design guidelines there are several UIs that you won't need for this exercise; also, some pages are not given a design and you will create them following the design guidelines of the other pages.

Original design idea by [Gregoire Vella on Behance](https://www.behance.net/gregoirevella).

The [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/) requires that you give appropriate credit to the author. Therefore, you must do it in the README of your project.

#### Interactions
- Splash screen
  - A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.

- Sign up and log in pages
  - The user should be able to register in the app with full name, email and password (all mandatory).
  - The user can log into the app using email and password.
  - If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

- Home page (categories page)
  - When the user logs in, they are presented with the categories page.
  - For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
  - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
  - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

- Transactions page
  - For a given category, the list of transactions is presented, ordered by the most recent.
  - At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
  - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new category" page
  - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
  - The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new transaction" page
  - The user fills out a form to create a new transaction with:
    - name (mandatory)
    - amount (mandatory)
    - categories (mandatory at least one)
  - The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
  - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

#### Testing requirements
- Create unit and integration tests for all the most important components of your RoR application.

#### Technical requirements

- You should use Postgres as your database.
- You should use devise for authentication.
- You should validate all user input to make sure that anyone with bad intentions cannot compromise your app.
- You can use a view template engine of your choice (.erb, .slim, .haml).
- The project should be deployed and accessible online.
- Your database schema should reflect the following structure:

> NOTE: do not change column names visible in the diagram. You need to change the "Entity"  name according to the theme you have chosen for your project (please note: "Transaction" is a name already used by ActiveRecord, so using it as a name for your model and table will result in an error).

### Project documentation

Once you have finished the development of the project, you should record a video presenting the features of the project you built. It is a video with a **maximum length of 5 minutes**. The content of the video should include:

- a description of the project.
- a demo of the project features.
- you should also highlight some interesting piece of code or something you built that you are very proud of.

Presentation video: [Video](https://www.loom.com/share/991a3985fba44c2cb665a4031f446450)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+## ⭐️ Show your support <a name="support"></a>
+
+If you like this project give me a follow and/or a star
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
+
+We would like to thank all my coding partners at Microverse
```
Comment (Abenezer-Tilahun) [inline README.md]: - [x] As stated in the project requirement Kindly make sure that The README contains an "Acknowledgements" section or similar that gives appropriate credit to [Gregoire Vella on Behance](https://www.behance.net/gregoirevella) the [author of the original design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), as required by the [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/).

File: app/views/transactions/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+        <%= f.label :name, "Name", class:"form-label" %><br>
+        <%= f.text_field :name,  :required => true, class:"form-control", placeholder: "Enter the name of the transaction" %>
+    </div>
+    <div class="mb-3">
+        <%= f.label :amount, "Amount", class:"form-label" %><br>
+        <%= f.number_field :amount,  :required => true, class:"form-control", placeholder: "How much did the transaction cost" %>
+    </div>
+    <button type="submit" class="btn btn-primary">Create</button>
+<%end%>
+
```
Comment (Abenezer-Tilahun) [inline app/views/transactions/new.html.erb]: - [x] As stated in the project requirement in the section of "Add a new transaction" page the user should fills out a form to **create a new transaction with: name (mandatory), amount (mandatory) and  categories (mandatory at least one) but In the case of yours the form contain name and amount so Kindly add categories at least one .**
Capture

**Review by Abenezer-Tilahun — CHANGES_REQUESTED** [review]
Hi @Mahamadou-Nouridine 

Great job for implemented  the some of the Project Requirement  , but there are some changes that need to be implemented in order to proceed to the next project 👍

## Highlights 👏🏻
✔️ No Linter errors
✔️ Correct Git-Flow
✔️ Good commit messages
✔️ Professional README
✔️ Correctly Config & basic setup Ruby on Rails to build a *a mobile web application*
✔️Correctly  design the Splash screen, Sign up and log in pages with validation.
✔️Correctly  design Home page (categories page) and "Add a new category" page with the specific requirement. 
✔️ Correctly  design Transactions page  with the specific requirement and Kindly adjust the design of "Add a new transaction" page based on the below mentioned requirement . 
✔️ Correctly Set up testing libraries(unit tests and integration tests for all the most important components of the RoR application) and All Test are passing.
✔️ Good Job for Implementing Technical Requirements and design according [design guidelines](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=),[design guidelines](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), 

almost

## Required Changes ♻️
_Check the comments below the review._

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/transactions/new.html.erb (hunk 2 above)
Comment (Mahamadou-Nouridine) [reply app/views/transactions/new.html.erb]: Thank you

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @Mahamadou-Nouridine👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- Nice UI design. ✔️
- Highly organized code. ✔️
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- No linter errors. :heavy_check_mark:
- PR has a good title and summary. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Mahamadou-Nouridine** [comment]
Thank you all @thecodechaser, @Abenezer-Tilahun

**Pull request merged**